vapi/sqlite3.vapi

```diff
@@ -0,0 +1,460 @@
+
+		public int wal_autocheckpoint (int N);
+		public int wal_checkpoint (string zDb);
+		public void* wal_hook (WALHookCallback cb, string db_name, int page_count);
+	}
+
+	[CCode (instance_pos = 0)]
+	public delegate int AuthorizeCallback (Sqlite.Action action, string? p1, string? p2, string db_name, string? responsible);
+	[CCode (instance_pos = 0)]
+	public delegate void TraceCallback (string message);
+	[CCode (instance_pos = 0)]
+	public delegate void ProfileCallback (string sql, uint64 time);
+	public delegate int ProgressCallback ();
+	public delegate int CommitCallback ();
+	public delegate void RollbackCallback ();
+	[CCode (has_target = false)]
+	public delegate void UserFuncCallback (Sqlite.Context context, [CCode (array_length_pos = 1.1)] Sqlite.Value[] values);
+	[CCode (has_target = false)]
+	public delegate void UserFuncFinishCallback (Sqlite.Context context);
+	[CCode (instance_pos = 0)]
+	public delegate void UpdateCallback (Sqlite.Action action, string dbname, string table, int64 rowid);
+	[CCode (instance_pos = 0)]
+	public delegate int CompareCallback (int alen, void* a, int blen, void* b);
+	[CCode (instance_pos = 0)]
+	public delegate int WALHookCallback (Sqlite.Database db, string dbname, int pages);
+
+	public unowned string? compileoption_get (int n);
+	public int compileoption_used (string option_name);
+	public static int complete (string sql);
+	[CCode (sentinel = "")]
+	public static int config (Sqlite.Config op, ...);
+	public unowned string libversion ();
+	public int libversion_number ();
+	[PrintfFormat]
+	public void log (int error_code, string format, ...);
+	public unowned string sourceid ();
+	public static int status (Sqlite.Status op, out int pCurrent, out int pHighwater, int resetFlag = 0);
+	public static int threadsafe ();
+
+	[CCode (cname = "SQLITE_VERSION")]
+	public const string VERSION;
+	[CCode (cname = "SQLITE_VERSION_NUMBER")]
+	public const int VERSION_NUMBER;
+	[CCode (cname = "SQLITE_SOURCE_ID")]
+	public const string SOURCE_ID;
+
+	/* Dynamically Typed Value Object */
+	[Compact]
+	[Immutable]
+	[CCode (cname = "sqlite3_value", copy_function = "sqlite3_value_dup", free_function = "sqlite3_value_free")]
+	public class Value {
+		[CCode (cname = "sqlite3_value_blob")]
+		public void* to_blob ();
+		[CCode (cname = "sqlite3_value_bytes")]
+		public int to_bytes ();
+		[CCode (cname = "sqlite3_value_double")]
+		public double to_double ();
+		[CCode (cname = "sqlite3_value_int")]
+		public int to_int ();
+		[CCode (cname = "sqlite3_value_int64")]
+		public int64 to_int64 ();
+		[CCode (cname = "sqlite3_value_text")]
+		public unowned string to_string ();
+		[CCode (cname = "sqlite3_value_type")]
+		public int to_type ();
+		[CCode (cname = "sqlite3_value_numeric_type")]
+		public int to_numeric_type ();
+		[CCode (cname = "sqlite3_value_dup")]
+		public Value copy ();
+	}
```

Comment:
This file is an exact copy of the sqlite3.vala distributed by valac, except I added `copy()` / `sqlite3_value_dup()`, and set the `copy_function`, `free_function`, and `[Immutable]`, to make it work with `Gee.List`

Reply:
Oh and I renamed `to_text()` to `to_string()`